www/apps/resources/app/recipes/marketplace/examples/vendors/page.mdx

```diff
@@ -0,0 +1,1344 @@
+    admin: adminData,
+  }: Pick<CreateVendorAdminWorkflowInput, "admin">, 
+  { container }) => {
+    const marketplaceModuleService: MarketplaceModuleService = 
+      container.resolve(MARKETPLACE_MODULE)
+
+    const vendorAdmin = await marketplaceModuleService.createVendorAdmins(
+      adminData
+    )
+
+    return new StepResponse(vendorAdmin)
+  }
+)
+
+export default createVendorAdminStep
+```
+
+This is the first step that creates the vendor admin and returns it.
+
+Then, create the workflow at `src/workflows/marketplace/create-vendor-admin/index.ts` with the following content:
+
+export const vendorAdminWorkflowHighlights = [
+  ["33", "setAuthAppMetadataStep", "Step is imported from `@medusajs/core-flows`."]
+]
+
+```ts title="src/workflows/marketplace/create-vendor-admin/index.ts" highlights={vendorAdminWorkflowHighlights}
+import { createWorkflow } from "@medusajs/workflows-sdk"
+import { 
+  setAuthAppMetadataStep,
+} from "@medusajs/core-flows"
+import createVendorAdminStep from "./steps/create-vendor-admin"
+
+export type CreateVendorAdminWorkflowInput = {
+  admin: {
+    email: string
+    first_name?: string
+    last_name?: string
+    vendor_id: string
+  }
+  authIdentityId: string
+}
+
+type CreateVendorAdminWorkflowOutput = {
+  id: string
+  first_name: string
+  last_name: string
+  email: string
+}
+
+const createVendorAdminWorkflow = createWorkflow<
+  CreateVendorAdminWorkflowInput, CreateVendorAdminWorkflowOutput
+>(
+  "create-vendor-admin",
+  function (input) {
+    const vendorAdmin = createVendorAdminStep({
+      admin: input.admin,
+    })
+
+    setAuthAppMetadataStep({
+      authIdentityId: input.authIdentityId,
+      actorType: "vendor",
+      value: vendorAdmin.id,
+    })
+
+    return vendorAdmin
+  }
+)
+
+export default createVendorAdminWorkflow
+```
+
+This runs the `createVendorAdminStep`, then the `setAuthAppMetadataStep` imported from `@medusajs/core-flows`, which creates the `vendor` actor type.
+
+The workflow returns the created vendor admin.
+
+### Further Read
+
+- [How to Create a Workflow](!docs!/basics/workflows)
+- [What is an Actor Type](../../../../commerce-modules/auth/auth-identity-and-actor-types/page.mdx)
+- [How to Create an Actor Type](../../../../commerce-modules/auth/create-actor-type/page.mdx)
+
+---
+
+## Step 5: Create Vendor API Route
+
+In this step, you’ll create the API route that runs the workflow from the previous step.
+
+Create the file `src/api/vendors/route.ts` with the following content:
+
+export const vendorRouteHighlights = [
+  ["10", "schema", "Define the fields expected in the request body."],
+  ["45", "parse", "Validate the request body and, if valid, retrieve it as an object."],
+  ["51", "createVendors", "Create the vendor using the Marketplace Module's main service."],
+  ["54", "createVendorAdminWorkflow", "Execute the workflow created in the first step."],
+]
+
+```ts title="src/api/vendors/route.ts" highlights={vendorRouteHighlights}
```

Comment:
thought(non-blocking): This file seems a bit overwhelming to get shown right away. I think we should consider breaking up the different elements of the file in a sort of incremental approach to creating the API route. Will make it more digestible.